### ISSUE TITLE
Pretrained Model

### ISSUE THREAD

**Issue by ZhangXH94** [zzh142857/SiameseFC-tf#1]
Hello,I am new in this field. Would you mind upload the pretrained model? Because I can run the tracker directly.Thank you anyway!

**Comment by zzh142857** [comment]
Hi,
I've upload the checkpoint file containing the model I trained. And if you just need to use the pretrained tracking to track a video, you can just use the model from the original author of this paper, see here: https://github.com/torrvision/siamfc-tf. My code is built on base of his code.

**Comment by ONeilFeng** [comment]
Hi,I tried to train a model from scratch using OTB videos. I used your hyper-parameters, but it seems that the loss remains between 0.40~0.50. I have run 68475 steps. So could you tell me your loss at last in your training process ? And what do you think may cause my situation now?
Thank you!

**Comment by zzh142857** [comment]
@ONeilFeng ,
Do you mean the loss is around 0.4-0.5 even at the beginning and doesn't decrease?  In my case, the initial loss is quite large,  and would eventually converge to 0.5. Sorry I have no clue what happened to your model, but I guess the first thing you can check is to plot the cropped image z and x to check if the code mess up the image and ground truth.  You can plot cropped z by uncommenting line 89-99 in src/trainer.py, and plot cropped x with the correspond ground truth response map by uncommenting line 132-139 in src.trainer.py. Please let me know if there is any update. I will keep track of the problem.

And one thing I have to admit, the code I posted is really a preliminary version. Although the model I trained has a not bad performance, bugs may still exist. And  I am still trying to optimize its performance. I will finalize it and post a better version recently.

**Comment by ONeilFeng** [comment]
@zzh142857 ,
Thank you for your timely and patient response. 
In my training process,  the initial loss is also quite large ,  around 6.0-7.0,  and eventually decreases and converges to 0.4-0.5.

**Comment by zzh142857** [comment]
@ONeilFeng, Then that kind of make sense I guess. Have you tried to test the model on any video to see the performance?

**Comment by ONeilFeng** [comment]
@zzh142857 I have not but I am just going to do that. Have you done that?

**Comment by zzh142857** [comment]
@ONeilFeng,Yes of course. It's not bad. The tracker can track lots of objects quite well. But not as good as the official version( https://github.com/torrvision/siamfc-tf). It makes sense, since I use OTB dataset which only has thousands of images, whereas the author uses millions of images to train the model, which is one of the main advantages of Siamese architecture, and also I didn't implement data augment for deformation，illumination change and other factors.

**Comment by ONeilFeng** [comment]
@zzh142857 
How many video do you think the author trained their model on? I think if all the videos are used in training process, the time will be dramatically long. Would you tell me how long did it take to train your model? I just want to make a comparison :)

**Comment by zzh142857** [comment]
@ONeilFeng, The author didn't use any video to train the model, however, his training data is generated from ImageNet image dataset. Since the training process of SiamFC only needs examplar z and candidates image x who both contain the same target. The training data doesn't need to be from a video. What the author did is select an image from an image dataset as examplar z, and generate the candidate image x by implementing some transformation to this image, such as deformation. Being able to use large amount image dataset instead of video sequence is one of the main advantage of SiamFC. Refer to part 2.3 and 2.4 in the original paper, the dataset the author used consists of 2 million images.

In my case, I used only 78 video sequences with total thousands of images. I found there exists severe over-fitting problems. It makes sense since the size of my training data is super small. And I found the tracker will have the best performance roughly around 5000 training step. So it actually takes less than a day to train it. And the checkpoint file in output/ folder I just uploaded today is trained by 5000 training steps. But this is only my personal experience, you should try it yourself. And then we can check one against another.

To do it in the correct way, one should use ImageNet image dataset which the author used to train this model. And the training process should takes several days I guess.
